tianshou/data/buffer.py

```diff
@@ -394,6 +480,17 @@ def reset(self) -> None:
             if isinstance(self._meta.__dict__[k], list):
                 self._meta.__dict__[k] = []
 
+    def __getstate__(self) -> dict:
+        return self.__dict__
```

Comment:
I don't think it is necessary because this is the original `__getstate__` function implementation. Maybe we need some extra tests to prove that?

Reply:
Yes, but no. It derives from ReplayBuffer, so it inherits its implementation. This if it it necessary to modify `set`, the same goes for `get`. Even though I agree it shouldn't be necessary to définie `set` in the first place 👍

Reply:
I put these methods in `ListReplayBuffer` and `PrioritizedReplayBuffer` to make them behave as before with pickle because I did not have a look at them yet. I tried to mention that above, sorry if I haven't made myself clear. Yes, in the final implementation it shouldn't be necessary to define these methods in derived classes.